cwms-data-api/src/main/java/cwms/cda/api/watersupply/WaterPumpDeleteController.java

```diff
@@ -0,0 +1,147 @@
+import static cwms.cda.api.Controllers.OFFICE;
+import static cwms.cda.api.Controllers.PROJECT_ID;
+import static cwms.cda.api.Controllers.WATER_USER;
+import static cwms.cda.data.dao.JooqDao.getDslContext;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import cwms.cda.api.errors.CdaError;
+import cwms.cda.data.dao.watersupply.WaterContractDao;
+import cwms.cda.data.dto.CwmsId;
+import cwms.cda.data.dto.watersupply.PumpType;
+import cwms.cda.data.dto.watersupply.WaterUserContract;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.http.HttpServletResponse;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.DSLContext;
+
+
+public final class WaterPumpDeleteController extends WaterSupplyControllerBase implements Handler {
+    private static final Logger LOGGER = Logger.getLogger(WaterPumpDeleteController.class.getName());
+    private static final String PUMP_TYPE = "pump-type";
+
+    public WaterPumpDeleteController(MetricRegistry metrics) {
+        waterMetrics(metrics);
+    }
+
+    @OpenApi(
+        queryParams = {
+            @OpenApiParam(name = PUMP_TYPE, required = true, type = PumpType.class,
+                    description = "The type of pump to be removed from the contract."
+                            + " Expected values: IN, OUT, OUT BELOW"),
+            @OpenApiParam(name = DELETE, type = boolean.class, required = true,
+                    description = "Whether to delete the associated accounting data.")
+        },
+        pathParams = {
+            @OpenApiParam(name = NAME, description = "The name of the pump to be "
+                    + "removed from the specified contract.", required = true),
+            @OpenApiParam(name = OFFICE, description = "The office the project is associated with.", required = true),
+            @OpenApiParam(name = PROJECT_ID, description = "The name of the project.", required = true),
+            @OpenApiParam(name = CONTRACT_NAME, description = "The name of the contract the pump is associated with.",
+                    required = true),
+            @OpenApiParam(name = WATER_USER, description = "The name of the water user the contract "
+                    + "is associated with.", required = true)
+        },
+        responses = {
+            @OpenApiResponse(status = "404", description = "The provided combination of parameters"
+                    + " did not find any contracts."),
+            @OpenApiResponse(status = "501", description = "Requested format is not implemented.")
+        },
+        description = "Delete a pump from a contract",
```

Comment:
language should specify "disassociate a pump from a contract" since it isn't deleting a pump

Reply:
Replaced references to "delete"